### ISSUE TITLE
Bug in nested API queries

### ISSUE THREAD

**Issue by Selutario** [wazuh/wazuh#19267]
|Wazuh version|Component|
|---|---|
| 4.5.3 | Wazuh API |

## Description

Support for nested queries in API requests was added in this PR:
- https://github.com/wazuh/wazuh/pull/18493

However, analyzing a [failed test in another PR](https://github.com/wazuh/wazuh/pull/18495/checks?check_run_id=17149283441), I have found that queries like these (for example in `GET /agents`):
```
(group=group2);(id<5;(os.major>14,os.major<20))
```

Are still failing:
```JSON
{
  "title": "Bad Request",
  "detail": "Error in wazuhdb request: Cannot execute Global database query; incomplete input",
  "remediation": "Make sure the request is correct",
  "dapi_errors": {
    "master-node": {
      "error": "Error in wazuhdb request: Cannot execute Global database query; incomplete input"
    }
  },
  "error": 2003
}
```

## Checks

The following elements have been updated or reviewed (should also be checked if no modification is required):
- [x] Tests (unit tests, API integration tests).
- [x] Changelog.
- [x] Documentation.
- [x] Integration test mapping (using `api/test/integration/mapping/_test_mapping.py`).

**Comment by GGP1** [comment]
## Update

The error mentioned ocurred in `4.7.1` and the changes from https://github.com/wazuh/wazuh/pull/18493 that were merged into `4.5.3` didn't reach that branch yet.

I made some tests to validate that the query is working as expected with the changes from the mentioned branch and it did.

<details><summary><code>(group=group2);(id<5;(os.major>14,os.major<20))</code></summary>

#### Filters

```
[
    {'value': 'group2', 'operator': '=', 'field': 'group$0', 'separator': 'AND', 'level': 0},
    {'value': '5', 'operator': '<', 'field': 'id$0', 'separator': 'AND', 'level': 1},
    {'value': '14', 'operator': '>', 'field': 'os.major$0', 'separator': 'OR', 'level': 2},
    {'value': '20', 'operator': '<', 'field': 'os.major$1', 'separator': '', 'level': 0}
]
```

#### SQL query

```sql
SELECT {0} FROM agent WHERE (group = :group$0 COLLATE NOCASE) AND ((id < :id$0 COLLATE NOCASE) AND ((os.major > :os_major$0 COLLATE NOCASE) OR (os.major < :os_major$1 COLLATE NOCASE)))
```

</details>

In fact, it's almost identical to the one that reported that nested queries weren't supported in the first place.

<details><summary><code>id!=000;(status=active;(group=default2,group=default3))</code></summary>

#### Filters

```
[
      {'value': '000', 'operator': '!=', 'field': 'id$0', 'separator': 'AND', 'level': 0},
      {'value': 'active', 'operator': '=', 'field': 'status$0', 'separator': 'AND', 'level': 1},
      {'value': 'default2', 'operator': '=', 'field': 'group$0', 'separator': 'OR', 'level': 2},
      {'value': 'default3', 'operator': '=', 'field': 'group$1', 'separator': '', 'level': 0}
]
```

#### SQL query

```sql
SELECT {0} FROM agent WHERE (id != :id$0 COLLATE NOCASE) AND ((status = :status$0 COLLATE NOCASE) AND ((group = :group$0 COLLATE NOCASE) OR (group = :group$1 COLLATE NOCASE)))
```

</details>

**Comment by fdalmaup** [comment]
# Review

I tested the mentioned behavior in `4.5.3` and `4.7.1`. Both branches worked fine and returned the expected response. Also, I compared the code from both versions and found that the changes introduced in #18493 were already merged in `4.7.1` during the upward merges for numbered branches (#19290) on September 28. 
The last update on the branch containing the failing checks was on October 2. Therefore, adding to the correct analysis done in this issue, which checked the internals of the feature, it does not seem that the origin of the error is in the nested queries.
